src/main/java/com/mixpanel/android/mpmetrics/MixpanelNotificationRouteActivity.java

```diff
@@ -75,59 +75,44 @@ protected Intent handleRouteIntent(Intent routeIntent) {
+        final String extraLogData = getStringFromBundle(intentExtras, "extraLogData");
+        final Boolean sticky = getBooleanFromBundle(intentExtras, "sticky");
 
-        final String tapTarget = tapTargetChars.toString();
         final String buttonId;
-        final String label;
-
-        if (tapTarget.equals(MixpanelPushNotification.TAP_TARGET_BUTTON)) {
-            CharSequence buttonIdChars = intentExtras.getCharSequence("buttonId");
-            if (null == buttonIdChars) {
-                MPLog.d(LOGTAG, "Notification action click logged with no buttonId");
-            }
-
-            CharSequence labelChars = intentExtras.getCharSequence("label");
-            if (null == labelChars) {
-                MPLog.d(LOGTAG, "Notification action click logged with no label");
-            }
-
-            buttonId = buttonIdChars == null ? null : buttonIdChars.toString();
-            label = labelChars == null ? null : labelChars.toString();
+        final String buttonLabel;
+        if (tapTarget != null && tapTarget.equals(MixpanelPushNotification.TAP_TARGET_BUTTON)) {
+            buttonId = getStringFromBundle(intentExtras, "buttonId");
+            buttonLabel = getStringFromBundle(intentExtras, "label");
         } else {
             buttonId = null;
-            label = null;
-        }
-
-        CharSequence messageIdChars = intentExtras.getCharSequence("messageId");
-        if (null == messageIdChars) {
-            MPLog.d(LOGTAG, "Notification action click logged with no messageId");
+            buttonLabel = null;
         }
 
-        CharSequence campaignIdChars = intentExtras.getCharSequence("campaignId");
-        if (null == campaignIdChars) {
-            MPLog.d(LOGTAG, "Notification action click logged with no campaignId");
-        }
-
-        final String messageId = messageIdChars == null ? null : messageIdChars.toString();
-        final String campaignId = campaignIdChars == null ? null : campaignIdChars.toString();
-
         MixpanelAPI.allInstances(new MixpanelAPI.InstanceProcessor() {
             @Override
             public void process(MixpanelAPI api) {
                 JSONObject pushProps = new JSONObject();
                 try {
-                    pushProps.put("tap_target", tapTarget);
-                    if (tapTarget.equals(MixpanelPushNotification.TAP_TARGET_BUTTON)) {
-                        pushProps.put("button_id", buttonId);
-                        pushProps.put("button_label", label);
+                    if (extraLogData != null) {
+                        pushProps = new JSONObject(extraLogData);
                     }
-                    pushProps.put("message_id", messageId);
-                    pushProps.put("campaign_id", campaignId);
+                } catch (JSONException e) {}
+                try {
+                    putIfNotNull(pushProps, "tap_target", tapTarget);
```

Comment:
shouldn't we reinitialize pushProps to a clean JSONObject if the try fails

Reply:
idk I imagine we'd like to provide "best effort" data